pkg/sql/tenant.go

```diff
@@ -231,7 +231,31 @@ func updateTenantRecord(
 }
 
 // CreateTenant implements the tree.TenantOperator interface.
-func (p *planner) CreateTenant(ctx context.Context, tenID uint64, name string) error {
+func (p *planner) CreateTenant(ctx context.Context, name string) (roachpb.TenantID, error) {
+	if err := p.RequireAdminRole(ctx, "create tenant"); err != nil {
+		return roachpb.SystemTenantID, err
+	}
+
+	// Find the next available ID that can be assigned to the created tenant.
+	row, err := p.execCfg.InternalExecutor.QueryRowEx(ctx, "next-tenant-id", p.Txn(), sessiondata.NodeUserSessionDataOverride, `
```

Comment:
@knz just confirming this query could result in us re-using a previously used ID, but this is okay because the tenant has to be GC'ed for us to be able to reuse the ID. 

Alternatively, we could change this query to just find the higher PK id and just use the next one.